### ISSUE TITLE
InvalidCastException on EnsureCreated() for MariaDB 10.3

### ISSUE THREAD

**Issue by proog** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#614]
### Steps to reproduce
Connect to a MariaDB 10.3 instance and call `EnsureCreated()` on a context when the database already exists. This can be an empty context.

```
static void Main(string[] args)
{
    var options = new DbContextOptionsBuilder<DbContext>()
        .UseMySql("Server=127.0.0.1;Uid=root;Database=pomelo").Options;
    new DbContext(options).Database.EnsureCreated();
}
```

### The issue
As of a few days ago, I've been hitting what appears to be the same issue as #600 - calling `EnsureCreated()` throws an `InvalidCastException`. Here's the stack trace for the above program when running it a second time (it works when the database does not already exist).

```
Unhandled Exception: System.InvalidCastException: Unable to cast object of type 'System.Int32' to type 'System.Int64'.
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlDatabaseCreator.<HasTables>b__13_0(IMySqlRelationalConnection connection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.MySqlExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Storage.RelationalDatabaseCreator.EnsureCreated()
   at proj.Program.Main(String[] args) in Program.cs:line 18
```

### Further technical details

This issue seems to be an incompatibility with MariaDB 10.3, or maybe just 10.3.7 (which I recently upgraded to). Downgrading to 10.2 (with a clean database) fixes the issue for me, and upgrading makes it reappear.

MySQL version: `Ver 10.3.7-MariaDB for osx10.13 on x86_64 (Homebrew)`
Operating system: OS X High Sierra
Pomelo.EntityFrameworkCore.MySql version: 2.1.0-rc1-final (same result with nightly)

**Comment by mguinness** [comment]
This appears to be where it's failing, but it's unclear why `ExecuteScalar` doesn't return a long instead of an int.  Probably could raise an issue in the underlying [MySqlConnector](https://github.com/mysql-net/MySqlConnector/issues) to determine what's occuring.

https://github.com/PomeloFoundation/Pomelo.EntityFrameworkCore.MySql/blob/7083319d9a8750104f010461a660e60ee9034117/src/EFCore.MySql/Storage/Internal/MySqlDatabaseCreator.cs#L70-L82

**Comment by jdmallen** [comment]
I too ran into this exact issue using MariaDB 10.3.8 and Pomelo 2.1.1. My stacktrace looks identical to OP and also points to the`HasTables` method. Glad it's not just me! I was going crazy knowing that none of my entities contained any longs.

Thanks for the tip on downgrading @proog; I'm going to give that a shot now.

**Comment by oobayly** [comment]
I've been bitten recently when upgrading MariaDB due to changes in default configuration parameters.

MariaDB list the changes, so normally it's a case of RTFM, but they're easy to miss.
https://mariadb.com/kb/en/library/changes-improvements-in-mariadb-103/
https://mariadb.com/kb/en/library/upgrading-from-mariadb-102-to-mariadb-103/

I'm going to have a look through and see if I can work out if it's a specific server variable that is causing the issue, and report back.

**Comment by caleblloyd** [comment]
fixed in #658
